src/consumer_api_service.ts

```diff
@@ -71,19 +72,27 @@ export class ConsumerApiService implements IConsumerApi {
     this._processModelService = processModelService;
   }
 
-  public onUserTaskWaiting(callback: Messages.CallbackTypes.OnUserTaskWaitingCallback): void {
+  public async onUserTaskWaiting(identity: IIdentity, callback: Messages.CallbackTypes.OnUserTaskWaitingCallback): Promise<void> {
+    await this._iamService.ensureHasClaim(identity, this._canSubscribeEvents);
```

Comment:
```suggestion
    await this._iamService.ensureHasClaim(identity, this._canSubscribeToEvents);
```

Warum nicht einfach einen Satz daraus machen?! Bitte auch bei dem zugehörigem Claim.